pkg/networking/networking.go

```diff
@@ -121,9 +126,27 @@ func (rt *defaultHeadersRoundTripper) RoundTrip(request *http.Request) (*http.Re
 
 	rt.logRoundTrip(newRequest, response, err)
 
+	if err != nil {
+		return nil, err
+	}
+
+	if rt.networkAccess.errorHandler != nil {
+		errFromStatus := errFromStatusCode(response.StatusCode)
```

Comment:
Suggestion: How about `errFromResponse()` this would enable to parse jsonapi errors into error catalog errors.